### ISSUE TITLE
Connection types that share a name

### ISSUE THREAD

**Issue by bcarroll22** [nuwave/laravel-graphql-relay#4]
Hey I've been trying to resolve an issue with the package and I was wondering if you had any ideas on how to solve it.

I have a user model, and the user has a connection called posts. I also have groups, and I want them to have a connection called posts also. When I create the connection 'posts' on both types, I get an exception stating that there's already a type in the schema called 'PostsConnection'.

Any way to remedy this off the top of your head? Thanks!

**Comment by chrissm79** [comment]
I actually came close to encountering this error without even knowing it. I have a "employees" connection on a Company and a "coworkers" connection on a User. They are both a list of the type User, I just decided to call them a different name.

I'll try to come up with a solution for this!

**Comment by chrissm79** [comment]
This one is a little involved. I'll extract the [egdeType and connectionType methods](https://github.com/nuwave/laravel-graphql-relay/blob/master/src/Types/RelayType.php#L137-L210) on `Nuwave\Relay\Types\RelayType` to their own classes (they are just ObjectTypes) to help out a bit when I can. In the meantime, you would need to create a [custom field](https://github.com/Folkloreatelier/laravel-graphql#custom-field), an edge type and a connection type and register them with GraphQL so we have a single instance of each. Then, in your UserType and GroupType you would do something like:

``` php
class UserType extends RelayType
{
    // ...

    /**
     * Available fields of Type.
     *
     * @return array
     */
    public function relayFields()
    {
        return [
            // ...
            'posts' => App\Http\GraphQL\Fields\PostsField::class
        ];
    }

    /**
     * List of related connections.
     *
     * @return array
     */
    public function connections()
    {
        return [
            // You would not put the posts connection here.
        ];
    }
}
```

The quick and dirty alternative would be to give each connection a unique name. Maybe "articles" for the user:

``` php
public function posts()
{
    return $this->hasMany(App\Post::class);
}

public function articles()
{
    return $this->posts();
}
```

**Comment by bcarroll22** [comment]
What I did for now was set a parent option in the attributes on the connection. I overloaded the RelayType function that creates connections, so now when it creates the connection between classes it does something like "UserPostConnection" instead of trying to create multiple "PostConnection".

I suppose what should really be happening is that everything that has posts should implement some sort of HasPosts interface. That seems like the way you'd be expected to do it with GraphQL to me, anyway.

**Comment by bcarroll22** [comment]
Based on the conversation we had in the other thread about the new relay package, I spend a while testing it out and came to the conclusion that for Laravel specifically, this package is superior in terms of functionality. In order to help move this package forward, I'll most likely try to move everything I've done back into the package, and submit PRs to add functionality where it's needed to make it work with Lumen, etc..

As far as coding style goes, would you prefer I submit all PRs to you in PSR-1 & 2 compliance? And, how do you feel about adding the capability to use this package with Lumen also?

Thanks again for the great package, and I look forward to hopefully working with you,

**Comment by chrissm79** [comment]
That sounds excellent @bcarroll22, I could use the help! It would be great if you could submit PRs in PSR-2 compliance and it would be even better if we could get this package could work with Lumen. I haven't used it myself yet, but it's nice to know that it could be an option.

Really looking forward to seeing how this package evolves! If you see any examples of Relay using Subscriptions (or other functionality we don't have here) let me know and we'll start integrating!

**Comment by bcarroll22** [comment]
Fixed in #7
